driver-async/src/main/com/mongodb/async/client/AggregateFluent.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright 2015 MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.mongodb.async.client;
+
+import com.mongodb.async.SingleResultCallback;
+
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Fluent interface for aggregate.
+ *
+ * @param <T> The type of the result.
+ * @since 3.0
+ */
+public interface AggregateFluent<T> extends MongoIterable<T> {
+
+    /**
+     * Enables writing to temporary files. A null value indicates that it's unspecified.
+     *
+     * @param allowDiskUse true if writing to temporary files is enabled
+     * @return this
+     * @mongodb.driver.manual reference/command/aggregate/ Aggregation
+     * @mongodb.server.release 2.6
+     */
+    AggregateFluent<T> allowDiskUse(final Boolean allowDiskUse);
+
+    /**
+     * Sets the maximum execution time on the server for this operation.
+     *
+     * @param maxTime  the max time
+     * @param timeUnit the time unit, which may not be null
+     * @return this
+     * @mongodb.driver.manual reference/method/cursor.maxTimeMS/#cursor.maxTimeMS Max Time
+     */
+    AggregateFluent<T> maxTime(final long maxTime, final TimeUnit timeUnit);
+
+    /**
+     * Sets whether the server should use a cursor to return results.
+     *
+     * @param useCursor whether the server should use a cursor to return results
+     * @return this
+     * @mongodb.driver.manual reference/command/aggregate/ Aggregation
+     * @mongodb.server.release 2.6
+     */
+    AggregateFluent<T> useCursor(final Boolean useCursor);
+
+    /**
+     * Aggregates documents according to the specified aggregation pipeline, which must end with a $out stage.
+     *
+     * @param callback the callback, which is called when the aggregation completes
+     * @mongodb.driver.manual aggregation/ Aggregation
+     */
+    void toCollection(SingleResultCallback<Void> callback);
+
+    /**
+     * Sets the number of documents to return per batch.
+     *
+     * @param batchSize the batch size
+     * @return this
+     * @mongodb.driver.manual reference/method/cursor.batchSize/#cursor.batchSize Batch Size
+     */
+    AggregateFluent<T> batchSize(int batchSize);
```

Comment:
This and all others like it can be annotated with Override, since it overrides the method in MongoIterable.